core/modules/xrdsvc/SsiProvider.cc

```diff
@@ -0,0 +1,148 @@
+// -*- LSST-C++ -*-
+/*
+ * LSST Data Management System
+ * Copyright 2014-2015 AURA/LSST.
+ *
+ * This product includes software developed by the
+ * LSST Project (http://www.lsst.org/).
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the LSST License Statement and
+ * the GNU General Public License along with this program.  If not,
+ * see <http://www.lsstcorp.org/LegalNotices/>.
+ */
+/// Implement XrdSsiProviderServer to provide Qserv's SsiService
+/// implementation. Link this file when building a plugin to be used as
+/// ssi.svclib or oss.statlib.
+
+// Class header
+#include "xrdsvc/SsiProvider.h"
+
+// System headers
+#include <memory>
+#include <sstream>
+#include <sys/types.h>
+
+// Third party headers
+#include "XrdSsi/XrdSsiCluster.hh"
+#include "XrdSsi/XrdSsiLogger.hh"
+
+// LSST headers
+#include "lsst/log/Log.h"
+
+// Qserv headers
+#include "global/ResourceUnit.h"
+#include "wpublish/ChunkInventory.h"
+#include "xrdsvc/XrdName.h"
+
+/******************************************************************************/
+/*                               G l o b a l s                                */
+/******************************************************************************/
+
+XrdSsiProvider *XrdSsiProviderServer = 
+                new  lsst::qserv::xrdsvc::SsiProviderServer;
+
+XrdSsiProvider *XrdSsiProviderLookup = XrdSsiProviderServer;
+  
+/******************************************************************************/
+/*                                  I n i t                                   */
+/******************************************************************************/
+  
+namespace lsst {
+namespace qserv {
+namespace xrdsvc {
+
+bool SsiProviderServer::Init(XrdSsiLogger* logP,  XrdSsiCluster* clsP, char const* cfgFn,
+          char const*   parms, int argc, char **argv) {
+
+    lsst::qserv::xrdsvc::XrdName x;
+
+// Establish our instance name (many different qservs may be running here).
+//
+    _name = x.getName();
+
+// Save the ssi logger as it places messages in anoher file than our log.
```

Comment:
Well, we can change that. All we need to do is come up with exactly how 
you want log messages to be routed and that can be built into the XRootD 
infrastructure. You are not the only person that has made that point. What 
no one has wanted to do is spend time to come up with the "rules".

Andy

On Wed, 14 Oct 2015, Andy wrote:

>> +/******************************************************************************/
>> +
>> +namespace lsst {
>> +namespace qserv {
>> +namespace xrdsvc {
>> +
>> +bool SsiProviderServer::Init(XrdSsiLogger* logP,  XrdSsiCluster* clsP, char const* cfgFn,
>> +          char const*   parms, int argc, char **argv) {
>> +
>> +    lsst::qserv::xrdsvc::XrdName x;
>> +
>> +// Establish our instance name (many different qservs may be running here).
>> +//
>> +    _name = x.getName();
>> +
>> +// Save the ssi logger as it places messages in anoher file than our log.
>
> For worker side (xrootd side) we have special log4cxx appender configured which redirects LOG_SMTHG() output to xrootd log (I'd prefer it other way round but xrootd logging is not message-oriented so it's hard to do). SO in principle for workers we should have just one log file. Or do I miss something else?
>
> ---
> Reply to this email directly or view it on GitHub:
> https://github.com/lsst/qserv/pull/143/files#r42085816